css-values-5/Overview.bs

```diff
@@ -549,7 +549,7 @@ Selecting the First Supported Value: the ''first-valid()'' notation</h3>
 	as the whole value of the property it's used in.
 
 	<pre class=prod>
-		<dfn>&lt;first-valid()></dfn> = first-valid( <<declaration-value>> [ ';' <<declaration-value>> ]* )
+		<dfn>&lt;first-valid()></dfn> = first-valid( <<declaration-value>>? [ ';' <<declaration-value>>? ]* )
```

Comment:
Nah, this was intended. The only place an empty value is valid is in untyped custom properties, and you can't actually use first-valid() in such properties (it's just seen as part of the substitution value, not evaluated).

Reply:
> you can't actually use `first-valid()` in such properties

I assume this will be added to the spec later, and that *untyped custom properties* means custom properties not registered with `@property`, or registered with the universal `syntax` value. 

Does it mean that `first-valid()` must be invalid for the `initial-value` of custom properties declared with the universal syntax?

  > (it's just seen as part of the substitution value, not evaluated)

Sorry, I do not understand.

Reply:
I mean that in `--foo: first-valid(foo, bar);`, the UA does not see a `first-valid()` function there. It just sees several uninterpreted tokens. The only syntax that is actually processed in an unregistered custom property is `var()`. The `first-valid()` won't be processed until it's substituted into either a real property or a registered custom property with a non-universal grammar. (Which is precisely the value here - we lost the ability to do `prop: old-val; prop: new-val;` if you're using vars - `--foo: old-val; --foo: new-val; prop: var(--foo);` doesn't work. But `--foo: first-valid(new-val, old-val); prop: var(--foo);` works like the first example.)

Reply:
`first-valid()` is perfectly fine for use in for custom properties; that's its *primary use-case*, in fact. It's just not useful *for custom properties themselves*, as in, for selecting the first valid that is valid *for the custom property itself*.

Reply:
Ah yes, it will be processed when substituting a property or a typed custom property. I clearly get the point about its usage now, thanks. I was bothered and stuck by the idea that an empty value cannot be a *first valid* value of an untyped custom property, whereas it is valid value according to its grammar.

I assume that untyped custom properties declared with `first-valid()` should always serialize unresolved, whereas other properties declared with `first-valid()` should serialize with the first valid value resolved at computed value time.

Reply:
Correct.